File: frontend/src/model/GameCardDetailsModel.ts
import {Genre} from "./Genre";
import {Publisher} from "./Publisher";
import {Platform} from "./Platform";

export type GameCardDetailsModel = {
    description_raw: string;
    id: number;
    title: string;
    releaseDate: string;
    publishers: Publisher[];
    genres: Genre[];
    cover: string;
    rating: number;
    platforms: Platform[];

}

File: backend/src/main/java/com/github/rohkaz/model/GameCardDetailsModel.java
package com.github.rohkaz.model;

import com.fasterxml.jackson.annotation.JsonAlias;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.List;

public record GameCardDetailsModel(

        @JsonProperty("description_raw") String description,
        int id,
        @JsonAlias("name") String title,
        @JsonAlias("released") String releaseDate,
        List<Publisher> publishers,
        List<Genre> genres,
        @JsonAlias("background_image") String cover,
        @JsonAlias("metacritic") int rating


) {
}
